### ISSUE TITLE
Ignores single-line /*! ... */ comment contents sometimes (Origin: bugzilla #327697)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#2006]
status NEW severity *normal* in component *general* for *---*
Reported in version *1.4.x* on platform *Other*
Assigned to: Dimitri van Heesch

Original attachment names and IDs:
- _[bug_327697.taz](https://sourceforge.net/projects/doxygen/files/attachments/bug327697_334752_bug_327697.taz/download/download)_ (ID 334752)

On 2006-01-19 13:08:06 +0000, Marc Mutz wrote:

    Sample C++ file (doxygen v1.4.6):
    
    ---foo.h---
    /*!
      \file foo.h The foo.h file
    */
    
    /*! This function performs the foo operation on \a i */
    void foo( int i ); // works
    
    /*! \overload */
    void foo( double d ); // ignored
    
    /*!
      \overload
    */
    void foo( double r, double i ); // works
    --end--
    
    I'll attach my Doxyfile.

On 2006-01-19 13:32:01 +0000, Marc Mutz wrote:

    Since 'Create a New Attachment' doesn't work for me from here, pasting
    instead grep -vE '^$|^#' Doxyfile
    
    --Doxyfile--
    CREATE_SUBDIRS         = NO
    OUTPUT_LANGUAGE        = English
    USE_WINDOWS_ENCODING   = NO
    BRIEF_MEMBER_DESC      = YES
    REPEAT_BRIEF           = YES
    ABBREVIATE_BRIEF       =
    ALWAYS_DETAILED_SEC    = NO
    INLINE_INHERITED_MEMB  = NO
    FULL_PATH_NAMES        = YES
    STRIP_FROM_PATH        =
    STRIP_FROM_INC_PATH    =
    SHORT_NAMES            = NO
    JAVADOC_AUTOBRIEF      = NO
    MULTILINE_CPP_IS_BRIEF = NO
    DETAILS_AT_TOP         = NO
    INHERIT_DOCS           = YES
    DISTRIBUTE_GROUP_DOC   = NO
    TAB_SIZE               = 8
    ALIASES                =
    OPTIMIZE_OUTPUT_FOR_C  = NO
    OPTIMIZE_OUTPUT_JAVA   = NO
    SUBGROUPING            = YES
    EXTRACT_ALL            = NO
    EXTRACT_PRIVATE        = YES
    EXTRACT_STATIC         = YES
    EXTRACT_LOCAL_CLASSES  = NO
    EXTRACT_LOCAL_METHODS  = NO
    HIDE_UNDOC_MEMBERS     = YES
    HIDE_UNDOC_CLASSES     = YES
    HIDE_FRIEND_COMPOUNDS  = NO
    HIDE_IN_BODY_DOCS      = NO
    INTERNAL_DOCS          = NO
    CASE_SENSE_NAMES       = YES
    HIDE_SCOPE_NAMES       = NO
    SHOW_INCLUDE_FILES     = YES
    INLINE_INFO            = YES
    SORT_MEMBER_DOCS       = NO
    SORT_BRIEF_DOCS        = NO
    SORT_BY_SCOPE_NAME     = NO
    GENERATE_TODOLIST      = YES
    GENERATE_TESTLIST      = YES
    GENERATE_BUGLIST       = YES
    GENERATE_DEPRECATEDLIST= YES
    ENABLED_SECTIONS       =
    MAX_INITIALIZER_LINES  = 30
    SHOW_USED_FILES        = YES
    SHOW_DIRECTORIES       = YES
    FILE_VERSION_FILTER    =
    QUIET                  = YES
    WARNINGS               = YES
    WARN_IF_UNDOCUMENTED   = NO
    WARN_IF_DOC_ERROR      = YES
    WARN_NO_PARAMDOC       = NO
    WARN_FORMAT            = "$file:$line: $text"
    WARN_LOGFILE           = doxygen.log
    FILE_PATTERNS          = *.cpp *.h
    RECURSIVE              = NO
    EXCLUDE                = qtshared/lane_data.cpp
    EXCLUDE_SYMLINKS       = NO
    EXCLUDE_PATTERNS       = moc_*.cpp ui_*.h
    EXAMPLE_PATH           =
    EXAMPLE_PATTERNS       =
    EXAMPLE_RECURSIVE      = NO
    IMAGE_PATH             =
    INPUT_FILTER           =
    FILTER_PATTERNS        =
    FILTER_SOURCE_FILES    = NO
    SOURCE_BROWSER         = NO
    INLINE_SOURCES         = NO
    STRIP_CODE_COMMENTS    = YES
    REFERENCED_BY_RELATION = YES
    REFERENCES_RELATION    = YES
    VERBATIM_HEADERS       = YES
    ALPHABETICAL_INDEX     = NO
    COLS_IN_ALPHA_INDEX    = 5
    IGNORE_PREFIX          =
    GENERATE_HTML          = YES
    HTML_OUTPUT            = html
    HTML_FILE_EXTENSION    = .html
    HTML_HEADER            =
    HTML_FOOTER            =
    HTML_STYLESHEET        =
    HTML_ALIGN_MEMBERS     = NO
    GENERATE_HTMLHELP      = NO
    CHM_FILE               =
    HHC_LOCATION           =
    GENERATE_CHI           = NO
    BINARY_TOC             = NO
    TOC_EXPAND             = NO
    DISABLE_INDEX          = NO
    ENUM_VALUES_PER_LINE   = 4
    GENERATE_TREEVIEW      = NO
    TREEVIEW_WIDTH         = 250
    GENERATE_LATEX         = YES
    LATEX_OUTPUT           = latex
    LATEX_CMD_NAME         = latex
    MAKEINDEX_CMD_NAME     = makeindex
    COMPACT_LATEX          = NO
    PAPER_TYPE             = a4wide
    EXTRA_PACKAGES         =
    LATEX_HEADER           =
    PDF_HYPERLINKS         = NO
    USE_PDFLATEX           = NO
    LATEX_BATCHMODE        = NO
    LATEX_HIDE_INDICES     = NO
    GENERATE_RTF           = NO
    RTF_OUTPUT             = rtf
    COMPACT_RTF            = NO
    RTF_HYPERLINKS         = NO
    RTF_STYLESHEET_FILE    =
    RTF_EXTENSIONS_FILE    =
    GENERATE_MAN           = NO
    MAN_OUTPUT             = man
    MAN_EXTENSION          = .3
    MAN_LINKS              = NO
    GENERATE_XML           = NO
    XML_OUTPUT             = xml
    XML_SCHEMA             =
    XML_DTD                =
    XML_PROGRAMLISTING     = YES
    GENERATE_AUTOGEN_DEF   = NO
    GENERATE_PERLMOD       = NO
    PERLMOD_LATEX          = NO
    PERLMOD_PRETTY         = YES
    PERLMOD_MAKEVAR_PREFIX =
    ENABLE_PREPROCESSING   = YES
    MACRO_EXPANSION        = NO
    EXPAND_ONLY_PREDEF     = NO
    SEARCH_INCLUDES        = YES
    INCLUDE_PATH           =
    INCLUDE_FILE_PATTERNS  =
    PREDEFINED             =
    EXPAND_AS_DEFINED      =
    SKIP_FUNCTION_MACROS   = YES
    TAGFILES               = qt.tag
    GENERATE_TAGFILE       =
    ALLEXTERNALS           = NO
    EXTERNAL_GROUPS        = YES
    PERL_PATH              = /usr/bin/perl
    CLASS_DIAGRAMS         = YES
    HIDE_UNDOC_RELATIONS   = YES
    HAVE_DOT               = NO
    CLASS_GRAPH            = YES
    COLLABORATION_GRAPH    = YES
    GROUP_GRAPHS           = YES
    UML_LOOK               = NO
    TEMPLATE_RELATIONS     = NO
    INCLUDE_GRAPH          = YES
    INCLUDED_BY_GRAPH      = YES
    CALL_GRAPH             = NO
    GRAPHICAL_HIERARCHY    = YES
    DIRECTORY_GRAPH        = YES
    DOT_IMAGE_FORMAT       = png
    DOT_PATH               =
    DOTFILE_DIRS           =
    MAX_DOT_GRAPH_WIDTH    = 1024
    MAX_DOT_GRAPH_HEIGHT   = 1024
    MAX_DOT_GRAPH_DEPTH    = 0
    DOT_TRANSPARENT        = NO
    DOT_MULTI_TARGETS      = NO
    GENERATE_LEGEND        = YES
    DOT_CLEANUP            = YES
    SEARCHENGINE           = NO
    ---end---
    

On 2006-01-31 12:45:37 +0000, Paul Fotheringham wrote:

    Confirmed that this example fails to produce the expected documentation
    using Doxygen 1.4.6 on Windows.
    
    Also discovered that the problem disappears if a newline is inserted after
    the \overload command in the comment block before void foo( double d );

On 2016-09-04 14:18:36 +0000, albert wrote:

    Created attachment 334752
    Example showing the problem
    
    I had a new look at the problem and saw that the original problem seems to
    be solved, in case \overload does not have an argument all versions are
    shown (see class myfoo).
    In case of having a reference to the function that is overloaded (see class
    myfoo1) the version where the \overload <function that is overloaded> is not
    the end of the comment (i.e. at least a newline follows the version is
    missing; in our case void foo1 (double d, int j)) is missing in the detailed
    description and an extra line is added to the <function that is overloaded>.
    Furthermore there are in the overloaded functions no references to the
    <function that is overloaded>.
    
    I looked at the code but could not find a solution, a few points I saw:
    - commentscan.l has 2 was to bail out of the <OverloadParam> state:
      - a newline, and we go into the <Comment> state.
      - End of File / end of block, we go back to the parseCommentBlock routine,
    where something is done for the <OverloadParam> state.
      Why is this not handled in one go in the <OverloadParam>{DOCNL} part by
    adding here <OverloadParam>EOF, we would close the <OverloadParam> state and
    go through the end of file state of <Comment> to end the entire comment.
    - doxygen.cpp, in the routine findMember, which has a parameter overloaded
    that is only TRUE in case of an OVERLOADDOC_SEC, there is the test "if
    (!className.isEmpty()) // class name is valid" which is valid even for an
    "overloaded" section. Extending this test to "if (!className.isEmpty() &&
    !overloaded) // class name is valid" is not sufficient as in that case in
    the documentation a "Friends And Related Function Documentation" section
    will appear and the "not documented function" will not show in the detailed
    section and there will be no reference to the "function that is overloaded"
    either.

**Comment by albert-github** [comment]
As mentioned in the comment of  2016-09-04 14:18:36 the original problem was solved in the mean time. In the documentation it was written:
>     The \overload command does not work inside a one-line comment.

though this restriction has been removed.

The comments about that the situation didn't work when using `\overload void foo(int)` was a misinterpretation of the documentation as the syntax should have been the syntax of the overloaded function (in this case `void foo(double d, int j)`) and not the syntax of the the function that was being overloaded (in this case `void foo(int i)`).

I've just pushed a proposed patch, pull request #9699 regarding correction and clarification of the documentation.
